utils/model_wrappers/langchain_chat_models.py

```diff
@@ -1136,6 +1194,341 @@ def __init__(self, **kwargs: Any) -> None:
+                )
+                # -> {
+                #     'raw': AIMessage(content='{\n  "answer": "They are the same weight",\n  "justification": "A pound is a unit of weight or mass, so a pound of bricks and a pound of feathers both weigh the same amount, one pound. The difference is in their density and volume. A pound of feathers would take up more space than a pound of bricks due to the difference in their densities."\n}', additional_kwargs={}, response_metadata={'finish_reason': 'stop', 'usage': {'acceptance_rate': 5.3125, 'completion_tokens': 79, 'completion_tokens_after_first_per_sec': 292.65701089829776, 'completion_tokens_after_first_per_sec_first_ten': 346.43324678555325, 'completion_tokens_per_sec': 200.012158915008, 'end_time': 1731528071.1708555, 'is_last_response': True, 'prompt_tokens': 70, 'start_time': 1731528070.737394, 'time_to_first_token': 0.16693782806396484, 'total_latency': 0.3949759876026827, 'total_tokens': 149, 'total_tokens_per_sec': 377.2381225105847}, 'model_name': 'Meta-Llama-3.1-70B-Instruct', 'system_fingerprint': 'fastcoe', 'created': 1731528070}, id='83208297-3eb9-4021-a856-ca78a15758df'),
+                #     'parsed': AnswerWithJustification(answer='They are the same weight', justification='A pound is a unit of weight or mass, so a pound of bricks and a pound of feathers both weigh the same amount, one pound. The difference is in their density and volume. A pound of feathers would take up more space than a pound of bricks due to the difference in their densities.'),
+                #     'parsing_error': None
+                # }
+
+        """  # noqa: E501
+        if kwargs:
+            raise ValueError(f'Received unsupported arguments {kwargs}')
+        is_pydantic_schema = _is_pydantic_class(schema)
+        if method == 'function_calling':
+            if schema is None:
+                raise ValueError("schema must be specified when method is 'function_calling'. " 'Received None.')
+            tool_name = convert_to_openai_tool(schema)['function']['name']
+            llm = self.bind_tools([schema], tool_choice=tool_name)
+            if is_pydantic_schema:
+                output_parser: OutputParserLike[Any] = PydanticToolsParser(
+                    tools=[schema],  # type: ignore[list-item]
+                    first_tool_only=True,
+                )
+            else:
+                output_parser = JsonOutputKeyToolsParser(key_name=tool_name, first_tool_only=True)
+        elif method == 'json_mode':
+            llm = self
+            # TODO bind response format when json mode available by API
+            # llm = self.bind(response_format={"type": "json_object"})
+            if is_pydantic_schema:
+                schema = cast(Type[BaseModel], schema)
+                output_parser = PydanticOutputParser(pydantic_object=schema)
+            else:
+                output_parser = JsonOutputParser()
+
+        elif method == 'json_schema':
```

Comment:
There doesn't seem to be much difference between json_schema and json_mode. It seems that you could skip json_schema altogether because all it does is check if schema is None